app/src/main/kotlin/com/mapbox/vision/teaser/replayer/ReplayModeFragment.kt

```diff
@@ -0,0 +1,188 @@
+                context = requireContext(),
+                basePath = sessionsPath,
+                clickSessionListener = { onSessionClick(it) },
+                clickCameraListener = { onCameraClick() },
+                onActivateMultiSelectionListener =  { onActivateMultiSelection() }
+            )
+            recycler_sessions.adapter = sessionsAdapter
+        } else {
+            requireActivity().onBackPressed()
+        }
+    }
+
+    private fun initSelectAllButton() {
+        select_all.setOnClickListener {
+            sessionsAdapter.apply {
+                selectAll()
+                val count = getSelectedCount()
+                replay_fragment_title.text = requireContext().resources.getQuantityString(R.plurals.selected_items, count, count)
+            }
+        }
+    }
+
+    private fun initEditButton() {
+        edit_sessions_list.setOnClickListener {
+            changeMultiSelection(true)
+        }
+    }
+
+    private fun initSwipeRefreshLayout() {
+        swipe_refresh_sessions.setOnRefreshListener {
+            sessionsAdapter.updateSessionsList()
+            swipe_refresh_sessions.isRefreshing = false
+        }
+    }
+
+    private fun initDoneEditButton() {
+        done_edit.setOnClickListener {
+            changeMultiSelection(false)
+        }
+    }
+
+    private fun onActivateMultiSelection() {
+        changeMultiSelection(true)
+        setMultiSelectionTitle()
+    }
+
+
+    private fun initDeleteSessionsButton() {
+        delete_sessions.setOnClickListener {
+            sessionsAdapter.let {
+                val sessionsPath = arguments?.getString(ARG_PARAM_SESSIONS_PATH)
+                if (it.isMultiSelection && sessionsPath != null) {
+                    for (session in it.selectedItems) {
+                        val sessionsFolder = File("$sessionsPath/$session")
+                        if (!sessionsFolder.deleteRecursively()) {
+                            Toast.makeText(requireContext(), "Error: Can't delete ${sessionsFolder.absoluteFile}", Toast.LENGTH_LONG).show()
+                        }
+                    }
+                    it.updateSessionsList()
+                    changeMultiSelection(false)
+                }
+            }
+        }
+    }
+
+    private fun onSessionClick(fileName: String) {
+        if (sessionsAdapter.isMultiSelection) {
+            setMultiSelectionTitle()
+        } else {
+            val listener = requireActivity()
+            if (listener is OnClickModeItemListener) {
+                listener.onClickSessionItem(fileName)
+                requireActivity().onBackPressed()
+            }
+        }
+    }
+
+    private fun onCameraClick() {
+        val listener = requireActivity()
+        if (listener is OnClickModeItemListener) {
+            listener.onClickCamera()
+            requireActivity().onBackPressed()
+        }
+    }
+
+    private fun setMultiSelectionTitle() {
+        sessionsAdapter.apply {
```

Comment:
What's the need for apply here? Seems
```
val count = sessionsAdapter.getSelectedCount()
replay_fragment_title.text = requireContext().resources.getQuantityString(R.plurals.selected_items, count, count)
```
is shorter and clearer

Reply:
fixed